src/io.jl

```diff
@@ -290,12 +292,16 @@ function save(
     @assert Z_NO_COMPRESSION <= compression_level <= Z_BEST_COMPRESSION
     @assert 2 <= ndims(image) <= 3
     @assert size(image, 3) <= 4
-    iswritable(s) || throw(ArgumentError("write failed, IOStream is not writeable"))
+    # iswritable(s) || throw(ArgumentError("write failed, IOStream is not writeable"))
 
     png_ptr = create_write_struct()
     info_ptr = create_info_struct(png_ptr)
     maybe_lock(s) do
-        png_set_write_fn(png_ptr, s.handle, writecallback_c[], C_NULL)
+        if s isa Base64EncodePipe
+            png_set_write_fn(png_ptr, s.io.data, writecallback_c[], C_NULL)
```

Comment:
actually the expected pipeline is: 

`array data -> png encoding -> Base64 encoding --IObuffer--> String`

what this does is:

`array data -> png encoding --IOBuffer--> String`

If we directly write to IOBuffer, then the data won't get Base64 encoding.

[This](https://github.com/JuliaLang/julia/blob/60c36f39cca453b6875ce8f530cfe0d5657daeae/stdlib/Base64/src/encode.jl#L202-L212) is the outer wrapper of this IO pipe, as you can see, data is written into IO Buffer and then fetched by `String(take!(io))`.

One way I haven't yet successfully to do is to allocate a temporary IO buffer to receive the data, and then pipe into the Base64EncodePiPe `s`, which is:

`array data -> png encoding --temp IOBuffer--> Base64EncodePipe --> String`